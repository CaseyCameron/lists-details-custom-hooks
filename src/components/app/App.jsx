import React from 'react';
import { Switch, Route } from 'react-router-dom';
import CharacterList from '../characters/CharacterList';
import CharacterDetails from '../details/CharacterDetail';

export default function App() {
  return (
    <Switch>
      <Route exact path="/" component={CharacterList} />
      <Route exact path="/:id" component={CharacterDetails} />
    </Switch>
  );
}