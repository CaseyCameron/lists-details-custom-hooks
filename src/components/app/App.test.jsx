import React from 'react';
import { render, screen, cleanup, waitFor } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';

describe('App component', () => {
  afterEach(() => cleanup());
  it('Displays a list of characters on the home page', async () => {
    render(
      <MemoryRouter initialEntries={['/']}>
        <App />
      </MemoryRouter>
    );
    screen.getByText('Loading...');
    const ul = await screen.findByRole('list');
    expect(ul).not.toBeEmptyDOMElement();
  });

  it('displays a characters details', async () => {
    render(
      <MemoryRouter initialEntries={['/5cf5679a915ecad153ab68d1']}>
        <App />
      </MemoryRouter>
    );

    await screen.findByText('Analay');
  });
});
