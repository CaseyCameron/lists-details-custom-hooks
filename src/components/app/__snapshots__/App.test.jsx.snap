// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div>
    <a
      href="/"
    >
      Avatar Characters Home
    </a>
    <h1>
      Loading...
    </h1>
  </div>
</DocumentFragment>
`;
